convert-sg-db.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python
+
+import json
+import commonmark
+from datetime import date
+
+
+def create_biblio(db):
+    biblio = {}
+
+    for workstream in db["workstreams"]:
+        for standard in workstream["standards"]:
+            reference, contents = standard_or_idea_to_biblio(standard)
+            contents["title"] += " Standard"
+            biblio[reference] = contents
+
+    for idea in db["ideas"]:
+        reference, contents = standard_or_idea_to_biblio(idea)
+        contents["status"] = "Living Idea"
+        biblio[reference] = contents
+
+    for biblio_item in db["biblio"]:
+        reference, contents = biblio_item_to_biblio(biblio_item)
+        biblio[reference] = contents
+
+    string = json.dumps(biblio, ensure_ascii=False, allow_nan=False, sort_keys=True, indent=2)
+    string += "\n"
+    open("resources.whatwg.org/biblio.json", "w").write(string)
+
+def standard_or_idea_to_biblio(document):
+    return (document["reference"], {
+        "authors": [author_to_biblio(author) for author in document["authors"]],
+        "href": document["href"],
+        "title": document["name"]
+    })
+
+def biblio_item_to_biblio(item):
+    contents = {
+        "href": item["href"],
+        "title": item["title"]
+    }
+    if len(item["authors"]):
+        contents["authors"] = [author_to_biblio(author) for author in item["authors"]]
+    if "obsoletedBy" in item:
+        contents["obsoletedBy"] = item["obsoletedBy"]
+    return (item["reference"], contents)
+
+def author_to_biblio(person):
+    return person["name"]
+
+
+def create_spec_whatwg_org(db, template):
+    content = "<p>The WHATWG works on a number of technologies that are fundamental parts of the web platform. They are organised somewhat arbitrarily based on the preferences of those editing the standard for those technologies.</p>\n\n"
+    content += "<dl>"
+    for workstream in db["workstreams"]:
+        for standard in workstream["standards"]:
+            content += standard_or_idea_to_html(standard)
+    content += "</dl>\n\n<p>The WHATWG also works on a <a href=\"https://idea.whatwg.org/\">number of ideas</a> that aspire to become standards one day.</p>\n\n"
+
+    template = template.replace("<a href=\"https://spec.whatwg.org/\">Standards</a>", "<a>Standards</a>")
+    write_site_template("spec.whatwg.org/index.html", template, "Standards", content)
+
+def standard_or_idea_to_html(document):
+    output = "\n <dt><a href=\"{0}\">{1}</a></dt>".format(document["href"], document["name"])
+    return output + "\n <dd>{}</dd>\n".format(commonmark.commonmark(document["description"])[:-1]) # Strip trailing \n
+
+def write_site_template(filename, template, title, content):
```

Comment:
Move this down above main to keep the `create_`s together?

Reply:
I put them in the order you encounter them in (except for main, which is at the bottom).

Reply:
I think that's significantly less helpful than logical ordering, and would really prefer otherwise.

Reply:
What is logical ordering?

Reply:
Grouping related things, especially things of the same inputs/return values, together. Instead of interspersing helpers between them.

Reply:
Is this documented somewhere? I looked at PEP8 and it only seems to care about import order.

Reply:
No, this isn't Python specific, it's just a general programming style thing. A quick Googling shows things like https://google.github.io/styleguide/cppguide.html#Declaration_Order or https://kotlinlang.org/docs/reference/coding-conventions.html#class-layout